kedro/io/partitioned_dataset.py

```diff
@@ -190,7 +193,8 @@ def __init__(  # pylint: disable=too-many-arguments
         self._filename_suffix = filename_suffix
         self._overwrite = overwrite
         self._protocol = infer_storage_options(self._path)["protocol"]
-        self._partition_cache: Cache = Cache(maxsize=1)
+        self._partition_cache = Cache(maxsize=1)  # type: Cache
```

Comment:
I've recently updated the code base to use a newer way of type annotation: https://github.com/kedro-org/kedro/pull/2538. We should use this syntax going forward. 
```suggestion
        self._partition_cache: Cache = Cache(maxsize=1)
```